### PR TITLE
Delete result native query

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

File: src/main/resources/db/changelog/changesets/changelog_20231123T101034Z.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-latest.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
+    <changeSet author="homereti" id="1700734364456-1">
```
Comment (etiennehomer) [inline src/main/resources/db/changelog/changesets/changelog_20231123T101034Z.xml:3]: Not sure that  it's necessary

File: src/main/java/org/gridsuite/shortcircuit/server/repositories/FaultResultRepository.java [hunk 2]
```diff
@@ -28,4 +26,17 @@ public interface FaultResultRepository extends JpaRepository<FaultResultEntity,
 
     @EntityGraph(attributePaths = {"feederResults"}, type = EntityGraphType.LOAD)
     Set<FaultResultEntity> findAllWithFeederResultsByFaultResultUuidIn(List<UUID> faultResultsUUID);
+
+    @Modifying
+    @Query(value = "DELETE FROM fault_result_entity WHERE result_result_uuid = ?1", nativeQuery = true)
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/shortcircuit/server/repositories/FaultResultRepository.java:31]: This request takes about 400ms for 7000 fault results to delete. 
While deleting feeder results lasts only 10ms over 25 000 results

**Review by ayolab — APPROVED** [review]

**Review by sBouzols — APPROVED** [review]
Code Review OK
Tests OK
5 delete requests for a single shortcircuit result deletion
some details for a classic network : 
```
gridstudy-postgres-1  | 2023-11-27 16:39:28.791 UTC [1204] LOG:  duration: 0.026 ms  plan:
gridstudy-postgres-1  | 	Query Text: delete from global_status where result_uuid=$1
--
gridstudy-postgres-1  | 2023-11-27 16:39:28.895 UTC [1204] LOG:  duration: 51.460 ms  plan:
gridstudy-postgres-1  | 	Query Text: DELETE FROM feeder_results WHERE fault_result_entity_fault_result_uuid IN ($1
--
gridstudy-postgres-1  | 2023-11-27 16:39:29.001 UTC [1204] LOG:  duration: 16.422 ms  plan:
gridstudy-postgres-1  | 	Query Text: DELETE FROM limit_violations WHERE fault_result_entity_fault_result_uuid IN ($1
--
gridstudy-postgres-1  | 2023-11-27 16:39:29.710 UTC [1204] LOG:  duration: 685.265 ms  plan:
gridstudy-postgres-1  | 	Query Text: DELETE FROM fault_result_entity WHERE result_result_uuid = $1
--
gridstudy-postgres-1  | 2023-11-27 16:39:29.728 UTC [1204] LOG:  duration: 1.451 ms  plan:
gridstudy-postgres-1  | 	Query Text: delete from shortcircuit_result where result_uuid=$1
```

File: src/main/java/org/gridsuite/shortcircuit/server/repositories/FaultResultRepository.java [hunk 3]
```diff
@@ -28,4 +26,17 @@ public interface FaultResultRepository extends JpaRepository<FaultResultEntity,
...
+    void deleteFaultResultsByShortCircuitResultUUid(UUID resultUuid);
+
+    @Modifying
+    @Query(value = "DELETE FROM limit_violations WHERE fault_result_entity_fault_result_uuid IN (SELECT fault_result_uuid FROM fault_result_entity where result_result_uuid = ?1)", nativeQuery = true)
+    void deleteLimitViolationsByShortCircuitResultUUid(UUID resultUuid);
+
+    @Modifying
+    @Query(value = "DELETE FROM feeder_results WHERE fault_result_entity_fault_result_uuid IN (SELECT fault_result_uuid FROM fault_result_entity where result_result_uuid = ?1)", nativeQuery = true)
+    void deleteFeederResultsByShortCircuitResultUUid(UUID resultUuid);
```
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/main/java/org/gridsuite/shortcircuit/server/repositories/FaultResultRepository.java]: will keep it here for now

**Pull request merged**